### PR TITLE
FEATURE: reduce avatar sizes to 6 from 20

### PR DESCRIPTION
**Pull request by SamSaffron**
This PR introduces 3 changes:

1. SiteSetting.avatar_sizes, now does what is says on the tin.
previously it would introduce a large number of extra sizes, to allow for
various DPIs. Instead we now trust the admin with the size list.

2. When `avatar_sizes` changes, we ensure consistency and remove resized
avatars that are not longer allowed per site setting. This happens on the
12 hourly job and limited out of the box to 20k cleanups per cycle, given
this may reach out to AWS 20k times to remove things.

3.Our default avatar sizes are now "24|48|72|96|144|288" these sizes were
very specifically picked to limit amount of bluriness introduced by webkit.
Our avatars are already blurry due to 1px border, so this corrects old blur.

This change heavily reduces storage required by forums which simplifies
site moves and more.

### CODE REVIEW

**Comment by SamSaffron** [comment]
@awesomerobot @davidtaylorhq now that specs are (hopefully) passing... where are we on merging this? 

I guess to be correct we want to rebake stuff with quotes, but CDN should still include old ones for the time being... I guess a migration to set cooked version to 0 for all posts with [quote in raw?

File: app/assets/javascripts/discourse/app/lib/utilities.js [hunk 1]
```diff
@@ -62,11 +62,32 @@ export function avatarUrl(template, size, { customGetURL } = {}) {
...
   const templatedPath = template.replace(/\{size\}/g, rawSize);
   return (customGetURL || getURLWithCDN)(templatedPath);
 }
 
+let allowedSizes = null;
+
+export function getRawAvatarSize(size) {
+  allowedSizes =
+    allowedSizes ||
+    helperContext()
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/lib/utilities.js:75]:
```suggestion
  allowedSizes ??=
    helperContext()
```
(`||=` would be technically equivalent, but I think the nullish coalescing operator better describes what we're doing here)

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -142,7 +142,8 @@ def user_error_message(*keys)
       end
 
       it "should not create any sidebar section link records for non human users" do
-        user = Fabricate(:user, id: -Time.now.to_i)
```
Comment (davidtaylorhq) [inline spec/models/user_spec.rb:145]: The changes in this test don't seem relevant to this PR - can we split it out?

**Review by davidtaylorhq — APPROVED** [review]
A few small comments, but otherwise looks great 👍

File: spec/models/badge_spec.rb [hunk 3]
```diff
@@ -44,11 +44,17 @@
   end
 
   it "can ensure consistency" do
-    b = Badge.first
```
[the remark this answers is not in the document]
Comment (SamSaffron) [reply spec/models/badge_spec.rb:47]: the tests are broken ... :( I guess we can ... but I don't want to block things on untangling
